### PR TITLE
Users prmittal summary

### PR DESCRIPTION
**Pull request by prmittal**

### CODE REVIEW

File: Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1 [hunk 1]
```diff
@@ -258,17 +271,45 @@ if ($drop.dropType -eq "InPlaceDrop")
...
+	
+    $summary = ('[Test Run: {0}]({2}/_apps/hub/ms.vss-cloudloadtest-web.hub-loadtest-account?_a=summary&runId={3}) using {1}.<br/>' -f  $run.runNumber, $run.name, $TFSAccountUrl, $run.id)
+    
+    $runComparisonAvailable = $false
+    $lastSuccessfulBuild = GetLastSuccessfultBuild $h
+	
+	if ($lastSuccessfulBuild)
+	{
+        $runSourceIdentifierFilter=('runsourceidentifier=build/{0}/{1}' -f $env:SYSTEM_DEFINITIONID, $lastSuccessfulBuild.id)
+        $runsInLastBuild = GetFilteredTestRuns $h $runSourceIdentifierFilter
```
Comment (sthangad) [inline Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1]: >$h [](start = 47, length = 2)

can be renamed as $headers for more readability

File: Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1 [hunk 2]
```diff
@@ -258,17 +271,45 @@ if ($drop.dropType -eq "InPlaceDrop")
...
     $summaryFile =  ("{0}\QuickPerfTestResults_{1}_{2}_{3}_{4}.md" -f $resultsMDFolder, $env:AGENT_ID, $env:SYSTEM_DEFINITIONID, $env:BUILD_BUILDID, $run.id)
+	
+    $summary = ('[Test Run: {0}]({2}/_apps/hub/ms.vss-cloudloadtest-web.hub-loadtest-account?_a=summary&runId={3}) using {1}.<br/>' -f  $run.runNumber, $run.name, $TFSAccountUrl, $run.id)
+    
+    $runComparisonAvailable = $false
+    $lastSuccessfulBuild = GetLastSuccessfultBuild $h
+	
+	if ($lastSuccessfulBuild)
+	{
+        $runSourceIdentifierFilter=('runsourceidentifier=build/{0}/{1}' -f $env:SYSTEM_DEFINITIONID, $lastSuccessfulBuild.id)
```
Comment (sthangad) [inline Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1]: >$runSourceIdentifierFilter [](start = 8, length = 26)

should we also add a timerange to be more performant.

File: Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1 [hunk 3]
```diff
@@ -258,17 +271,45 @@ if ($drop.dropType -eq "InPlaceDrop")
...
     $resultsMDFolder = New-Item -ItemType Directory -Force -Path "$env:Temp\LoadTestResultSummary"
     $resultFilePattern = ("QuickPerfTestResults_{0}_{1}_*.md" -f $env:AGENT_ID, $env:SYSTEM_DEFINITIONID)
     $excludeFilePattern = ("QuickPerfTestResults_{0}_{1}_{2}_*.md" -f $env:AGENT_ID, $env:SYSTEM_DEFINITIONID, $env:BUILD_BUILDID)
     Remove-Item $resultsMDFolder\$resultFilePattern -Exclude $excludeFilePattern -Force
     $summaryFile =  ("{0}\QuickPerfTestResults_{1}_{2}_{3}_{4}.md" -f $resultsMDFolder, $env:AGENT_ID, $env:SYSTEM_DEFINITIONID, $env:BUILD_BUILDID, $run.id)
+	
+    $summary = ('[Test Run: {0}]({2}/_apps/hub/ms.vss-cloudloadtest-web.hub-loadtest-account?_a=summary&runId={3}) using {1}.<br/>' -f  $run.runNumber, $run.name, $TFSAccountUrl, $run.id)
+    
+    $runComparisonAvailable = $false
+    $lastSuccessfulBuild = GetLastSuccessfultBuild $h
```
Comment (sthangad) [inline Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1]: >GetLastSuccessfultBuild [](start = 27, length = 23)

is this the right way to get the completed build..? won't be have the info available as a task context or something ?

**Comment by sthangad** [comment]
:clock1:

File: Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1 (hunk 3 above)
Comment (prmittal) [inline Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1]: @sthangad I did not find information related to the previous build in the current build's metadata.

---
In reply to: [74956393](https://github.com/Microsoft/vsts-tasks/pull/2265#discussion_r74956393) [](ancestors = 74956393)

File: Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1 (hunk 2 above)
Comment (prmittal) [inline Tasks/QuickPerfTest/Invoke-QuickPerfTest.ps1]: build definition with load test generally are not expected to run on a daily basis. Restriction of time is not a good option there.

---
In reply to: [74956185](https://github.com/Microsoft/vsts-tasks/pull/2265#discussion_r74956185) [](ancestors = 74956185)

**Comment by dpksinghal** [comment]
:shipit:

**Pull request merged**